port/port_modifier.go

```diff
@@ -16,47 +16,73 @@ package port
 
 import (
 	"encoding/json"
+	"fmt"
 	"net"
 	"net/http"
 	"strconv"
 	"strings"
 
-	"github.com/google/martian"
 	"github.com/google/martian/parse"
 )
 
 func init() {
 	parse.Register("port.Modifier", modifierFromJSON)
 }
 
-// Modifier alters the request URL and Host header to explictly
-// use the provided port.  In the case that the provided port is
-// the default port for the schema, the port will be explictly
-// declared.
+// Modifier alters the request URL and Host header to use the provided port.
+// Only one of port, defaultForScheme, or remove may be specified.  Whichever is set last is the one that will take effect.
+// If remove is true, remove the port from the host string ('example.com').
+// If defaultForScheme is true, explicitly specify 80 for HTTP or 443 for HTTPS ('http://example.com:80'). Do nothing for a scheme that is not 'http' or 'https'.
+// If port is specified, explicitly add it to the host string ('example.com:1234').
+// If port is zero and the other fields are false, the request will not be modified.
 type Modifier struct {
-	port int
+	port             int
+	defaultForScheme bool
+	remove           bool
 }
 
 type modifierJSON struct {
-	Port  int                  `json:"port"`
-	Scope []parse.ModifierType `json:"scope"`
+	Port             int                  `json:"port"`
+	DefaultForScheme bool                 `json:"defaultForScheme"`
+	Remove           bool                 `json:"remove"`
+	Scope            []parse.ModifierType `json:"scope"`
 }
 
-// NewModifier returns a RequestModifier that alters the request URL and Host header to explictly
-// use the provided port.  In the case that the provided port is
-// the default port for the schema, the port will be explictly
-// declared.
-func NewModifier(port int) martian.RequestModifier {
-	return &Modifier{
-		port: port,
-	}
+// NewModifier returns a RequestModifier that can be configured to alter the request URL and Host header's port.
+// One of DefaultPortForScheme, UsePort, or RemovePort should be called to configure this modifier.
+func NewModifier() *Modifier {
+	return &Modifier{}
+}
+
+// Explicitly specify 80 for HTTP or 443 for HTTPS ('http://example.com:80'). Do nothing for a scheme that is not 'http' or 'https'.
+// This overrides any previous configuration for this modifier.
+func (m *Modifier) DefaultPortForScheme() {
+	m.defaultForScheme = true
+	m.remove = false
+}
+
+// Explicitly add the specified port to the host string ('example.com:1234').
```

Comment:
The first sentence should be a one-sentence summary that starts with the name being declared.

https://golang.org/doc/effective_go.html#commentary